visual-dotnet/SauceLabs.Visual/VisualCheckOptions.cs

```diff
@@ -11,10 +11,23 @@ namespace SauceLabs.Visual
     /// </summary>
     public class VisualCheckOptions
     {
+        private FullPageConfig? _fullPageConfig;
         public DiffingMethod? DiffingMethod { get; set; }
         public IgnoreRegion[]? IgnoreRegions { get; set; }
         public IWebElement[]? IgnoreElements { get; set; }
         public bool? CaptureDom { get; set; }
+        public bool? FullPage { get; set; }
+        public FullPageConfig? FullPageConfig
+        {
+            get
+            {
+                return _fullPageConfig ?? (FullPage == true ? new FullPageConfig() : null);
```

Comment:
That logic should be in `VisualClient`.

`public FullPageConfig? FullPageConfig { get; set; }`

Then `VisualClient` will take care of dealing with that behaviour.

Reply:
Why it should be in `VisualClient`? If you will need to check if FullPage is turned on in other place you will duplicate the code.

Reply:
I also think that logic is related to `VisualCheckOptions`